src/main/java/spoon/support/compiler/jdt/PositionBuilder.java

```diff
@@ -691,7 +691,7 @@ private int getEndOfComment(char[] content, int maxOff, int off) {
 					if (content[off] == '\r') {
 						//we have found end of this comment
 						//skip windows \n too if any
-						if (content[off] == '\n') {
+						if (off < maxOff && content[off + 1] == '\n') {
```

Comment:
Yes, it needs + 1. But also in first part of condition. Like this:
`if (off + 1 < maxOff && content[off + 1] == '\n')`
OK?

Reply:
But why? The maximum value of `maxOff` is `content.length - 1` (see the upper lines), so it seems like a valid bounds check to me.

Reply:
You are right! So it is OK. Thank you.